Src/ILGPU/Util/NativeLibrary.cs

```diff
@@ -0,0 +1,120 @@
+﻿// ---------------------------------------------------------------------------------------
```

Comment:
What are the benefits of making the change?

Reply:
Currently, we use several classes generated at compile time that in turn use virtual methods to call the actual native methods. This means extra work at compile time to generate the various classes and additional overhead to invoke the virtual methods at runtime. This could be considerably simplified by using the `NativeLibrary` utility 🔢